### PR TITLE
Clippy pedantic

### PR DESCRIPTION
**Pull request by torhovland**
Minor improvements as suggested by `cargo clippy --all-targets --all-features -- -W clippy::pedantic`.

### CODE REVIEW

File: topiary/src/error.rs [hunk 1]
```diff
@@ -103,8 +103,7 @@ impl fmt::Display for FormatterError {
 
             Self::Internal(message, _)
             | Self::Query(message, _)
-            | Self::Io(IoError::Filesystem(message, _))
-            | Self::Io(IoError::Generic(message, _)) => {
+            | Self::Io(IoError::Filesystem(message, _) | IoError::Generic(message, _)) => {
```
Comment (Xophmeister) [inline topiary/src/error.rs:106]: TIL you could do that :+1:

**Review by Xophmeister — APPROVED** [review]
I'm not 100% convinced that the alternation-style in `match` branches, so that the RHS mapping is unique, is always better -- e.g., the error mapping to source, as it puts the enum members out of alphabetical order -- but you can't argue with consistency, I suppose!

Otherwise, I think these are pretty good improvements. Would it be worth adding those Clippy flags to the `default.nix`?

**Comment by torhovland** [comment]
> I'm not 100% convinced that the alternation-style in `match` branches, so that the RHS mapping is unique, is always better -- e.g., the error mapping to source, as it puts the enum members out of alphabetical order

Yes, I do see your point.

> Would it be worth adding those Clippy flags to the `default.nix`?

No, they are *too* pedantic. But I see now that we get even more suggestions by running:

```
cargo clippy --all-targets --all-features -- -W clippy::pedantic -W clippy::nursery -W clippy::cargo
```

**Review by Xophmeister — APPROVED** [review]
Looks like good changes to me :+1:

**Pull request merged**